src/backend/utils/adt/agtype.c

```diff
@@ -2377,7 +2378,9 @@ static agtype_value* agtype_build_map_as_agtype_value(FunctionCallInfo fcinfo)
     nargs = extract_variadic_args(fcinfo, 0, true, &args, &types, &nulls);
 
     if (nargs < 0)
-        PG_RETURN_NULL();
+    {
+        return NULL;
```

Comment:
Should this be `PG_RETURN_NULL()`? This macro sets `fcinfo->isnull = true;` before returning `0`.

Reply:
Good Question